kentik_api_library/kentik_api/api_connection/api_connector.py

```diff
@@ -39,13 +38,10 @@ def __init__(
     ) -> None:
         self._api_url = api_url
         self._logger = logging.getLogger(__name__)
-        self._session = RetryableSession(retry_strategy=retry_strategy)
-        self._session.auth = KentikAuth(auth_email, auth_token)
-        self._session.headers.update({"Content-Type": "application/json"})
+        self._session = prepare_kentik_api_http_session(auth_email, auth_token, retry_strategy, proxy)
```

Comment:
Session is constructed in api_transport_http and in api_connector; seems reasonable to keep just one copy of the code.
Not sure, what the gRPC api_connector would look like? Where would it be inserted? What it would be used for? api_connector for http was introduced for testing deserialization between json <-> actual data structures.

Reply:
In my view of the world, the `api_connector` handles connecting to the API end-point, sending requests, receiving responses, retries, timeouts and other transport level aspects. The cleanest solution would be for the upper layers (i.e. user visible objects) to not have to care about which transport is in use. I'm though not insisting on this level of abstraction since we do not need the transport to be selected at runtime. We need to support only 1 transport alternative for every API; specifically all (old) v5 APIs (devices, interfaces, ...) must keep using HTTP/REST and (new) v6 APIs (synthetics, cloudexport, mkp, ...) should use gRPC.

Reply:
The grpc generated code is the actual API client class. In current approach, it is going to be wrapped into "transport" class that abstracts library user from grpc or http. Should I rename transport to api_connector?

Reply:
As we are dropping http support, the current "transport" that causes confusion will go away

Reply:
updated.